contracts/src/v0.1/L1EndpointRequestResponse.sol

```diff
@@ -0,0 +1,301 @@
+// SPDX-License-Identifier: MIT
+pragma solidity ^0.8.16;
+import "./RequestResponseConsumerBase.sol";
+import "./L1EndpointBase.sol";
+import "./RequestResponseConsumerFulfill.sol";
+
+abstract contract L1EndpointRequestResponse is
+    RequestResponseConsumerBase,
+    L1EndpointBase,
+    RequestResponseConsumerFulfillUint128,
+    RequestResponseConsumerFulfillInt256,
+    RequestResponseConsumerFulfillBool,
+    RequestResponseConsumerFulfillString,
+    RequestResponseConsumerFulfillBytes32,
+    RequestResponseConsumerFulfillBytes
+{
+    using Orakl for Orakl.Request;
+
+    event DataRequested(uint256 requestId, address sender);
+    event DataRequestFulfilled(
+        uint256 requestId,
+        uint256 l2RequestId,
+        address sender,
+        uint256 callbackGasLimit,
+        bytes32 jobId,
+        uint128 responseUint128,
+        int256 responseInt256,
+        bool responseBool,
+        string responseString,
+        bytes32 responseBytes32,
+        bytes responseBytes
+    );
+
+    constructor(
+        address requestResponseCoordinator
+    ) RequestResponseConsumerBase(requestResponseCoordinator) {}
+
+    function requestData(
```

Comment:
`requestData` and `requestRandomWords` are very similar. Please consider creating auxiliary function(s) to reuse code from both of them.

Reply:
I think there are 2 different kind of requests, one with Keyhash and one without keyhash. But sure we can do a simple check to consolidate the 2.

Reply:
I understand. I am not talking about the whole function, but about the processes inside. There is check whether it is fulfilled by oracle, then we check whether it has enough balance and then we estimate fee. We don't have to it, it was just a suggestion.

Reply:
how about creating a separate issue for this. We need to do a grand clean up anyway

Reply:
Okay, the only thing that I still don't understand are those predefined requests in other comment. I will be more specific there.